projections/gorm.go

```diff
@@ -282,10 +305,107 @@ func sort(order map[string]string) func(db *gorm.DB) *gorm.DB {
 	}
 }
 
+//DateTimeChecks checks to make sure the format is correctly as well as it manipulates the date
+func DateTimeCheck(entityFactory weos.EntityFactory, properties map[string]FilterProperty) (map[string]FilterProperty, error) {
+	var err error
+	schema := entityFactory.Schema()
+	for key, value := range properties {
+		if schema.Properties[key] != nil && schema.Properties[key].Value.Format == "date-time" {
+			_, err := time.Parse(time.RFC3339, value.Value)
+			if err != nil {
+				return nil, err
+			}
+			//count := 1
+			//if value.Operator == "lt" {
+			//	timeH := timeP.Add(time.Duration(-count) * time.Second)
+			//	filter := FilterProperty{
+			//		Field:    properties[key].Field,
+			//		Operator: properties[key].Operator,
+			//		Value:    timeH.String(),
+			//	}
+			//	properties[key] = filter
+			//}
+			//if value.Operator == "gt" {
+			//	timeH := timeP.Add(time.Duration(count) * time.Second)
+			//	filter := FilterProperty{
+			//		Field:    properties[key].Field,
+			//		Operator: properties[key].Operator,
+			//		Value:    timeH.String(),
+			//	}
+			//	properties[key] = filter
+			//}
+
+		}
+	}
+
+	return properties, err
+}
+
+//filterStringBuilder is used to build the query strings
+func filterStringBuilder(dbName string, properties map[string]FilterProperty) string {
```

Comment:
why are you building the query manually and not using form for or and and conditions https://gorm.io/docs/query.html#Or-Conditions (you will run into issues when we try to run this across database types). It might be cleaner to make a Gorm Scope where you pass in the filterOptions and then in there update the db query with the and options and checking the operator to set what goes in the where clause e.g. ` .Where("name LIKE ?", "group%")`